eth/db/storage.py

```diff
@@ -129,9 +150,9 @@ def get_changed_root(self) -> Hash32:
             raise ValidationError("Asked for changed root when no writes have been made")
 
     def _clear_changed_root(self) -> None:
+        self._starting_root_hash = self._write_trie.root_hash
```

Comment:
If you get back in here could you maybe drop a comment or two in this method explaining in some detail why we do each of these things.  Since it's not clear how to address some of the debt in the code, maybe we can address some of with with better comments.